shared_model/interfaces/iroha_internal/transaction_batch_parser_impl.cpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "interfaces/iroha_internal/transaction_batch_parser_impl.hpp"
+
+#include <boost/range/adaptor/indirected.hpp>
+#include <boost/range/combine.hpp>
+#include "interfaces/iroha_internal/batch_meta.hpp"
+#include "interfaces/transaction.hpp"
+
+namespace {
+  template <typename InRange, typename OutRange>
+  auto parseBatchesImpl(InRange in_range, const OutRange &out_range) {
```

Comment:
Hard to read... What are these parameters? Maybe leave some comments here as well?